skdecide/hub/domain/gym/gym.py

```diff
@@ -700,30 +702,50 @@ def _discretize_action_space(
         self, action_space: gym.spaces.Space
     ) -> D.T_agent[Space[D.T_event]]:
         if isinstance(action_space, gym.spaces.box.Box):
-            nb_elements = 1
-            for dim in action_space.shape:
-                nb_elements *= dim
-            actions = []
-            for l, h in np.nditer([action_space.low, action_space.high]):
-                if l == -float("inf") or h == float("inf"):
-                    actions.append(
-                        [
-                            gym.spaces.box.Box(low=l, high=h).sample()
-                            for i in range(self._discretization_factor)
+            ticks = []
+
+            unbounded = ~action_space.bounded_below & ~action_space.bounded_above
+            upp_bounded = ~action_space.bounded_below & action_space.bounded_above
+            low_bounded = action_space.bounded_below & ~action_space.bounded_above
+            bounded = action_space.bounded_below & action_space.bounded_above
+
+            it = np.nditer(action_space.low, flags=["multi_index"])
+            for p in it:
```

Comment:
Variable `p` is not accessed, is it intended? If yes, please use another iteration scheme.

Reply:
@fteicht modification & rebase done. Tests are all good.